File: matiasmasso/Mat.NET/DT2/DTOBrandArea.vb
Public Class DTOBrandArea
    Property Guid As Guid
    Property Brand As DTOProductBrand
    Property Area As DTOArea
    Property FchFrom As Date
    Property FchTo As Date

    Property IsNew As Boolean
    Property IsLoaded As Boolean

    Shared Function Factory(oBrand As DTOProductBrand) As DTOBrandArea
        Dim retval As DTOBrandArea = Nothing
        If oBrand IsNot Nothing Then
            retval = New DTOBrandArea
            With retval
                .Guid = System.Guid.NewGuid
                .IsNew = True
                .Brand = oBrand
                .FchFrom = DateTime.Today
            End With
        End If
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOBrandArea.cs
using System;

namespace DTO
{
    public class DTOBrandArea
    {
        public Guid Guid { get; set; }
        public DTOProductBrand Brand { get; set; }
        public DTOArea Area { get; set; }
        public DateTime FchFrom { get; set; }
        public DateTime FchTo { get; set; }

        public bool IsNew { get; set; }
        public bool IsLoaded { get; set; }

        public static DTOBrandArea Factory(DTOProductBrand oBrand)
        {
            DTOBrandArea retval = null;
            if (oBrand != null)
            {
                retval = new DTOBrandArea();
                {
                    var withBlock = retval;
                    withBlock.Guid = System.Guid.NewGuid();
                    withBlock.IsNew = true;
                    withBlock.Brand = oBrand;
                    withBlock.FchFrom = DTO.GlobalVariables.Today();
                }
            }
            return retval;
        }
    }
}
